components/post_view/post_list/index.jsx

```diff
@@ -0,0 +1,449 @@
+         */
+        channelId: PropTypes.string,
+
+        /**
+         * Whether to display the channel intro at full width
+         */
+        fullWidth: PropTypes.bool,
+
+        /**
+         * Used for disabling loading more messages
+         */
+        postVisibility: PropTypes.number,
+
+        /**
+         * Used for showing more message indicators and channel intro message
+         */
+        channelPostsStatus: PropTypes.object,
+
+        /**
+         * Used for unread count
+         */
+        currentUserId: PropTypes.string,
+
+        /**
+         * Used for unread count
+         */
+        lastViewedAt: PropTypes.number,
+
+        /**
+         * Flag used to determine if channel has to call sync actions.
+         */
+        channelSyncStatus: PropTypes.bool,
+
+        /**
+         * Used for websocket connect change to trigger sync.
+         */
+        socketStatus: PropTypes.object,
+
+        /**
+         * Used for determining if we reached to the top of channel.
+         * Used for taking a backup before clearing from redux-store.
+         */
+        postIdsInCurrentChannel: PropTypes.array,
+
+        /*
+         * Object from react-router
+         */
+        match: PropTypes.shape({
+            params: PropTypes.shape({
+                messageId: PropTypes.string,
+            }).isRequired,
+        }).isRequired,
+
+        actions: PropTypes.shape({
+
+            /*
+             * Get post for permaview
+             */
+            getPostThread: PropTypes.func.isRequired,
+
+            /*
+             * Get unreads posts onload
+             */
+            loadUnreads: PropTypes.func.isRequired,
+
+            /*
+             * Get posts using BEFORE_ID and AFTER_ID
+             */
+            loadPosts: PropTypes.func.isRequired,
+
+            /*
+             * Used for clearing posts from channel
+             */
+            clearPostsFromChannel: PropTypes.func.isRequired,
+
+            /*
+             * Used for changing posts status of channel
+             * For notifying redux store when we hit bottom or top of posts
+             */
+            changeChannelPostsStatus: PropTypes.func.isRequired,
+
+            /*
+             * Used for taking backup of postsInChannel before clearing so,
+             * we can use it incase of failure in loading posts
+             */
+            backUpPostsInChannel: PropTypes.func.isRequired,
+
+            /*
+             * Used for notifying redux store on channel sync complete
+             */
+            channelSyncCompleted: PropTypes.func.isRequired,
+
+            /*
+             * Used for adding back postIds to postsInChannel from backUpPostsInChannel
+             */
+            addPostIdsFromBackUp: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            isDoingInitialLoad: this.shouldLoadPosts(props),
```

Comment:
This is on mount of the component. It is to prevent showing loader when posts are already present. This is a fairly inexpensive as it is just comparing few flags so, i don't think this should be a problem